cdap-securestore-ext-gcp-secretstore/src/main/java/io/cdap/cdap/securestore/gcp/cloudsecretmanager/WrappedSecret.java

```diff
@@ -0,0 +1,117 @@
+ * the License.
+ */
+
+package io.cdap.cdap.securestore.gcp.cloudsecretmanager;
+
+import com.google.cloud.secretmanager.v1.Replication;
+import com.google.cloud.secretmanager.v1.Replication.Automatic;
+import com.google.cloud.secretmanager.v1.Secret;
+import com.google.common.reflect.TypeToken;
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+import com.google.protobuf.util.Timestamps;
+import io.cdap.cdap.securestore.spi.secret.SecretMetadata;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Wrapper around a {@link io.cdap.cdap.securestore.spi.secret.SecretMetadata} which allows it to
+ * easily be converted a GCP SecretManager {@link com.google.cloud.secretmanager.v1.Secret} and
+ * back.
+ *
+ * <p>Does not contain the underlying secret payload.
+ */
+public final class WrappedSecret {
+  private static final Gson gson = new Gson();
+
+  private final String namespace;
+  private final SecretMetadata secretMetadata;
+
+  private WrappedSecret(String namespace, SecretMetadata secretMetadata) {
+    this.namespace = namespace;
+    this.secretMetadata = secretMetadata;
+  }
+
+  /** Constructs a new WrappedSecret from a CDAP Secret. */
+  public static WrappedSecret fromMetadata(String namespace, SecretMetadata metadata) {
+    return new WrappedSecret(namespace, metadata);
+  }
+
+  /** Constructs a new WrappedSecret from a GCP Secret Manager Secret. */
+  public static WrappedSecret fromGcpSecret(Secret secret) throws IOException {
+    String namespace = getNamespace(secret);
+    SecretMetadata metadata = toSecretMetadata(secret);
+    return new WrappedSecret(namespace, metadata);
+  }
+
+  public String getNamespace() {
+    return namespace;
+  }
+
+  public SecretMetadata getCdapSecretMetadata() {
+    return secretMetadata;
+  }
+
+  /**
+   * Returns a new GCP {@link Secret} representing the underlying SecretMetadata.
+   *
+   * @param resourceName Value to set for the "name" field needed for update operations.
+   */
+  public Secret getGcpSecret(String resourceName) {
+    return Secret.newBuilder()
+        // Set replication policy to automatic (as opposed to user-managed) and do not specify a
+        // CMEK (use google-managed key).
+        .setReplication(Replication.newBuilder().setAutomatic(Automatic.getDefaultInstance()))
+        .setName(resourceName)
+        .putAnnotations("cdap_namespace", namespace)
+        .putAnnotations("cdap_secret_name", secretMetadata.getName())
+        .putAnnotations(
+            "cdap_description", Optional.ofNullable(secretMetadata.getDescription()).orElse(""))
+        .putAnnotations("cdap_props", serializeProps(secretMetadata.getProperties()))
+        .build();
+  }
+
+  private static SecretMetadata toSecretMetadata(Secret secret) throws IOException {
+
+    return new SecretMetadata(
+        /* name= */ secret.getAnnotationsOrDefault("cdap_secret_name", ""),
```

Comment:
nit: we don't normally have comments at the start of the line, a little surprised checkstyle is not complaining. In any case, for consistency it's probably better to leave this out, as many IDEs already have settings that will display this information for people anyway.

Reply:
ack that IDEs should do this for us -- sadly code repos don't tend to do this for us so I have a habit of doing this.

updated.